### ISSUE TITLE
DatePicker & DateTimePicker: Unable to navigate through all elements using Tab key.

### ISSUE THREAD

**Issue by olehnoskov-cengage** [cengage/react-magma#2368]
User can't navigate navigate through all elements using Tab key after DatePicker & DateTimePicker components. The issue arose after `4.12.0` release.

[Code example](https://codesandbox.io/p/sandbox/wp2wvt)

https://cengage.slack.com/archives/C8Y4FR63U/p1774357009777279

**Comment by nikitaorliak-cengage** [comment]
Need to revert #2346 (only `useFocusLock` changes) and double-check it

**Comment by nikitaorliak-cengage** [comment]
For now, everything works as expected. We will try to fix this issue with MacOS + VO + Safari again.

**Comment by nikitaorliak-cengage** [comment]
The PR with this issue was reverted. New issue for `useFocusLock` is here -- https://github.com/cengage/react-magma/issues/2345
